app/assets/javascripts/discourse/app/mixins/password-validation.js

```diff
@@ -33,18 +33,21 @@ export default Mixin.create({
     "rejectedPasswords.[]",
     "accountUsername",
     "accountEmail",
-    "passwordMinLength"
+    "passwordMinLength",
+    "forceValidationReason"
   )
   passwordValidation(
     password,
     passwordRequired,
     rejectedPasswords,
     accountUsername,
     accountEmail,
-    passwordMinLength
+    passwordMinLength,
+    forceValidationReason
```

Comment:
same comment about the number of properties, could be in another PR

Reply:
I agree this is a lot of params, and I would do this in another PR, but what would you suggest doing with these? This and the other computed property are doing complex validations across multiple fields. I don't know how I could group them into one property, and this password validation mixin is used in a couple of different places.

What would your approach be if you had to do it?

Reply:
You could try to group things in state objects:

```
const options = {
  passwordRequired,
  passwordMinLength,
  rejectedPasswords
}

const account = {
  email,
  username
}
```

Reply:
Noting for posterity I am going to look into this in another PR, as it will probably require some refactoring.

Reply:
After taking a look at this I do not think it is worth the effort of refactoring it right now. Take `emailValidation`; it relies on `accountEmail` along with a few other properties, so then I put those into an `emailData` object the computed property ends up looking like this anyway, so I am not sure why this is better?

```javascript
    // Check the email address
    @discourseComputed(
      "emailData.{accountEmail,rejectedEmails,serverEmailValidation,serverAccountEmail}",
      "forceValidationReason"
    )
    emailValidation(emailData, forceValidationReason) {
```

`accountEmail` is then referenced by username-validation and password-validation, as these mixins are used together, which in turn is then used by both create-account.hbs and invites-show.hbs. invites-show.hbs also uses its own `emailValidation` function which copies a lot but not all of the stuff from create-account.js, further complicating things.

Just seems like a lot of churn for not much benefit to me, so I am leaving this for now. @jjaffeux